### DISCUSSION TITLE
Kiali-Operator v1.34.0  StartUp failure

### DISCUSSION THREAD

**Discussion by bethmage** [kiali/kiali#4004] [General]
Problem:
deploy the kiali_cr  with deployment.**accessible_namespaces:  '^((?!(kube.*|kiali-operator)).)*$'**
cause failure and does not create the kiali Pod  as excepted

Versions used
Kiali-Operator:
V1.34.0

Istio: 1.9.4
minikube version: v1.20.0  K8S v1.19.7  with  containerD

To Reproduce

install clean istio
pull and Install hhelm kiali-operator
```
helm upgrade --install --set cr.create=false --set cr.namespace=istio-system --set allowAdHocKialiImage=true  --namespace kiali-operator --create-namespace --namespace kiali-operator  --cleanup-on-fail kiali-operator ./kiali-operator-1.34.0.tgz
```
wait for helm deployment to be finish

create kiali_cr manifest:

```
cat <<EOF | kubectl -n istio-system  apply -f -
apiVersion: kiali.io/v1alpha1
kind: Kiali
metadata:
  name: kiali
  namespace: istio-system
  annotations:
    ansible.sdk.operatorframework.io/verbosity: "1"
spec:
  api:
    namespaces:
      label_selector: kiali-enabled=true
      exclude:
      - kube.*
      - kiali-operator
  auth:
    strategy: token
  deployment:
    image_version: v1.34.0
    accessible_namespaces:
    - '^((?!(kube.*|kiali-operator)).)*$'
  external_services:
    prometheus:
      auth:
        type: none
      url: http://prometheus-server.monitoringbackend:80
EOF
```

Expected behavior
kiali Pod should  be created and startUP

Namespace List:

```
NAME                STATUS   AGE     LABELS
access-psql         Active   8h      istio-injection=enabled
default             Active   25h     <none>
istio-system        Active   25h     istio-injection=disabled
kiali-operator      Active   25h     name=kiali-operator
kube-node-lease     Active   25h     <none>
kube-public         Active   25h     <none>
kube-system         Active   25h     <none>
monitoring          Active   8h      istio-injection=enabled
monitoringbackend   Active   7h32m   name=monitoringbackend

```
Logs:
kubectl -n istio-system describe  kialis.kiali.io kiali

```
Name:         kiali
Namespace:    istio-system
Labels:       <none>
Annotations:  ansible.sdk.operatorframework.io/verbosity: 1
API Version:  kiali.io/v1alpha1
Kind:         Kiali
Metadata:
  Creation Timestamp:  2021-05-11T16:59:45Z
  Finalizers:
    kiali.io/finalizer
  Generation:  1
  Managed Fields:
    API Version:  kiali.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .:
          v:"kiali.io/finalizer":
      f:status:
        .:
        f:conditions:
    Manager:      ansible-operator
    Operation:    Update
    Time:         2021-05-11T16:59:45Z
    API Version:  kiali.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .:
          f:ansible.sdk.operatorframework.io/verbosity:
          f:kubectl.kubernetes.io/last-applied-configuration:
      f:spec:
        .:
        f:api:
          .:
          f:namespaces:
            .:
            f:exclude:
            f:label_selector:
        f:auth:
          .:
          f:strategy:
        f:deployment:
          .:
          f:accessible_namespaces:
          f:image_version:
        f:external_services:
          .:
          f:prometheus:
            .:
            f:auth:
              .:
              f:type:
            f:url:
    Manager:      kubectl-client-side-apply
    Operation:    Update
    Time:         2021-05-11T16:59:45Z
    API Version:  kiali.io/v1alpha1
    Fields Type:  FieldsV1
    fieldsV1:
      f:status:
        f:progress:
          .:
          f:duration:
          f:message:
    Manager:         OpenAPI-Generator
    Operation:       Update
    Time:            2021-05-11T16:59:48Z
  Resource Version:  64577
  Self Link:         /apis/kiali.io/v1alpha1/namespaces/istio-system/kialis/kiali
  UID:               00c05999-fbe5-4d94-b893-7874a4cbfb75
Spec:
  API:
    Namespaces:
      Exclude:
        kube.*
        kiali-operator
      label_selector:  kiali-enabled=true
  Auth:
    Strategy:  token
  Deployment:
    accessible_namespaces:
      ^((?!(kube.*|kiali-operator)).)*$
    image_version:  v1.34.0
  external_services:
    Prometheus:
      Auth:
        Type:  none
      URL:     http://prometheus-server.monitoringbackend:80
Status:
  Conditions:
    Ansible Result:
      Changed:             4
      Completion:          2021-05-11T16:59:57.906762
      Failures:            1
      Ok:                  53
      Skipped:             59
    Last Transition Time:  2021-05-11T16:59:58Z
    Message:               Failed to patch object: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":" \\"\\" is invalid: patch: Invalid value: \\"{\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"kind\\\\\\":\\\\\\"Namespace\\\\\\",\\\\\\"metadata\\\\\\":{\\\\\\"annotations\\\\\\":{\\\\\\"kubectl.kubernetes.io/last-applied-configuration\\\\\\":\\\\\\"{\\\\\\\\\\\\\\"apiVersion\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"v1\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"kind\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"Namespace\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"metadata\\\\\\\\\\\\\\":{\\\\\\\\\\\\\\"annotations\\\\\\\\\\\\\\":{},\\\\\\\\\\\\\\"name\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"access-psql\\\\\\\\\\\\\\"}}\\\\\\\\n\\\\\\"},\\\\\\"creationTimestamp\\\\\\":\\\\\\"2021-05-11T08:15:22Z\\\\\\",\\\\\\"labels\\\\\\":{\\\\\\"istio-injection\\\\\\":\\\\\\"enabled\\\\\\",\\\\\\"kiali-enabled\\\\\\":true},\\\\\\"managedFields\\\\\\":[{\\\\\\"manager\\\\\\":\\\\\\"kubectl-client-side-apply\\\\\\",\\\\\\"operation\\\\\\":\\\\\\"Update\\\\\\",\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"time\\\\\\":\\\\\\"2021-05-11T08:15:22Z\\\\\\",\\\\\\"fieldsType\\\\\\":\\\\\\"FieldsV1\\\\\\",\\\\\\"fieldsV1\\\\\\":{\\\\\\"f:metadata\\\\\\":{\\\\\\"f:annotations\\\\\\":{\\\\\\".\\\\\\":{},\\\\\\"f:kubectl.kubernetes.io/last-applied-configuration\\\\\\":{}}},\\\\\\"f:status\\\\\\":{\\\\\\"f:phase\\\\\\":{}}}},{\\\\\\"manager\\\\\\":\\\\\\"kubectl-label\\\\\\",\\\\\\"operation\\\\\\":\\\\\\"Update\\\\\\",\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"time\\\\\\":\\\\\\"2021-05-11T11:07:14Z\\\\\\",\\\\\\"fieldsType\\\\\\":\\\\\\"FieldsV1\\\\\\",\\\\\\"fieldsV1\\\\\\":{\\\\\\"f:metadata\\\\\\":{\\\\\\"f:labels\\\\\\":{\\\\\\".\\\\\\":{},\\\\\\"f:istio-injection\\\\\\":{}}}}}],\\\\\\"name\\\\\\":\\\\\\"access-psql\\\\\\",\\\\\\"resourceVersion\\\\\\":\\\\\\"57673\\\\\\",\\\\\\"uid\\\\\\":\\\\\\"7884869d-3bff-4bef-8080-a77b157860e3\\\\\\"},\\\\\\"spec\\\\\\":{\\\\\\"finalizers\\\\\\":[\\\\\\"kubernetes\\\\\\"]},\\\\\\"status\\\\\\":{\\\\\\"phase\\\\\\":\\\\\\"Active\\\\\\"}}\\": v1.Namespace.ObjectMeta: v1.ObjectMeta.Labels: ReadString: expects \\" or n, but found t, error found in #10 byte of ...|enabled\\":true},\\"mana|..., bigger context ...|ls\\":{\\"istio-injection\\":\\"enabled\\",\\"kiali-enabled\\":true},\\"managedFields\\":[{\\"manager\\":\\"kubectl-client-s|...","reason":"Invalid","details":{"causes":[{"reason":"FieldValueInvalid","message":"Invalid value: \\"{\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"kind\\\\\\":\\\\\\"Namespace\\\\\\",\\\\\\"metadata\\\\\\":{\\\\\\"annotations\\\\\\":{\\\\\\"kubectl.kubernetes.io/last-applied-configuration\\\\\\":\\\\\\"{\\\\\\\\\\\\\\"apiVersion\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"v1\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"kind\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"Namespace\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"metadata\\\\\\\\\\\\\\":{\\\\\\\\\\\\\\"annotations\\\\\\\\\\\\\\":{},\\\\\\\\\\\\\\"name\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"access-psql\\\\\\\\\\\\\\"}}\\\\\\\\n\\\\\\"},\\\\\\"creationTimestamp\\\\\\":\\\\\\"2021-05-11T08:15:22Z\\\\\\",\\\\\\"labels\\\\\\":{\\\\\\"istio-injection\\\\\\":\\\\\\"enabled\\\\\\",\\\\\\"kiali-enabled\\\\\\":true},\\\\\\"managedFields\\\\\\":[{\\\\\\"manager\\\\\\":\\\\\\"kubectl-client-side-apply\\\\\\",\\\\\\"operation\\\\\\":\\\\\\"Update\\\\\\",\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"time\\\\\\":\\\\\\"2021-05-11T08:15:22Z\\\\\\",\\\\\\"fieldsType\\\\\\":\\\\\\"FieldsV1\\\\\\",\\\\\\"fieldsV1\\\\\\":{\\\\\\"f:metadata\\\\\\":{\\\\\\"f:annotations\\\\\\":{\\\\\\".\\\\\\":{},\\\\\\"f:kubectl.kubernetes.io/last-applied-configuration\\\\\\":{}}},\\\\\\"f:status\\\\\\":{\\\\\\"f:phase\\\\\\":{}}}},{\\\\\\"manager\\\\\\":\\\\\\"kubectl-label\\\\\\",\\\\\\"operation\\\\\\":\\\\\\"Update\\\\\\",\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"time\\\\\\":\\\\\\"2021-05-11T11:07:14Z\\\\\\",\\\\\\"fieldsType\\\\\\":\\\\\\"FieldsV1\\\\\\",\\\\\\"fieldsV1\\\\\\":{\\\\\\"f:metadata\\\\\\":{\\\\\\"f:labels\\\\\\":{\\\\\\".\\\\\\":{},\\\\\\"f:istio-injection\\\\\\":{}}}}}],\\\\\\"name\\\\\\":\\\\\\"access-psql\\\\\\",\\\\\\"resourceVersion\\\\\\":\\\\\\"57673\\\\\\",\\\\\\"uid\\\\\\":\\\\\\"7884869d-3bff-4bef-8080-a77b157860e3\\\\\\"},\\\\\\"spec\\\\\\":{\\\\\\"finalizers\\\\\\":[\\\\\\"kubernetes\\\\\\"]},\\\\\\"status\\\\\\":{\\\\\\"phase\\\\\\":\\\\\\"Active\\\\\\"}}\\": v1.Namespace.ObjectMeta: v1.ObjectMeta.Labels: ReadString: expects \\" or n, but found t, error found in #10 byte of ...|enabled\\":true},\\"mana|..., bigger context ...|ls\\":{\\"istio-injection\\":\\"enabled\\",\\"kiali-enabled\\":true},\\"managedFields\\":[{\\"manager\\":\\"kubectl-client-s|...","field":"patch"}]},"code":422}\n'
    Reason:                Failed
    Status:                False
    Type:                  Failure
    Last Transition Time:  2021-05-11T16:59:58Z
    Message:               Running reconciliation
    Reason:                Running
    Status:                True
    Type:                  Running
  Progress:
    Duration:  0:00:08
    Message:   2. Deleting obsolete roles
Events:        <none>

```

Kiali-Operator log.  TOO Big to put in here !

```
{"level":"info","ts":1620753445.0588057,"logger":"cmd","msg":"Version","Go Version":"go1.15.7","GOOS":"linux","GOARCH":"amd64","ansible-operator":"v1.3.0-ocp","commit":"2cd576d470160059d2d102bfc0b5153afe32a05a"}
{"level":"info","ts":1620753445.0589783,"logger":"cmd","msg":"Watching all namespaces.","Namespace":""}
{"level":"info","ts":1620753445.7676542,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1620753445.7691922,"logger":"ansible-controller","msg":"Watching resource","Options.Group":"kiali.io","Options.Version":"v1alpha1","Options.Kind":"Kiali"}
{"level":"info","ts":1620753445.7702475,"logger":"proxy","msg":"Starting to serve","Address":"127.0.0.1:8888"}
{"level":"info","ts":1620753445.7716234,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
{"level":"info","ts":1620753445.772011,"logger":"controller-runtime.manager.controller.kiali-controller","msg":"Starting EventSource","source":"kind source: kiali.io/v1alpha1, Kind=Kiali"}
{"level":"info","ts":1620753445.8729832,"logger":"controller-runtime.manager.controller.kiali-controller","msg":"Starting Controller"}
{"level":"info","ts":1620753445.8734798,"logger":"controller-runtime.manager.controller.kiali-controller","msg":"Starting workers","worker count":4}

{"level":"info","ts":1620753522.173602,"logger":"logging_event_handler","msg":"[playbook debug]","name":"kiali","namespace":"istio-system","gvk":"kiali.io/v1alpha1, Kind=Kiali","event_type":"runner_on_ok","job":"6129484611666145821","EventData.TaskArgs":""}

--------------------------- Ansible Task StdOut -------------------------------

 TASK [debug] ********************************
ok: [localhost] => {
    "msg": "KIALI RECONCILIATION START"
}

-------------------------------------------------------------------------------

...............  more ............
```

**Comment by jmazzitelli** [comment]
I see you are not using the helm chart that is published by the kiali team (https://kiali.org/helm-charts/index.yaml):
```
helm upgrade --install --set cr.create=false --set cr.namespace=istio-system --set allowAdHocKialiImage=true  --namespace kiali-operator --create-namespace --namespace kiali-operator  --cleanup-on-fail kiali-operator ./kiali-operator-1.34.0.tgz
```
We cannot validate this because we do not know where this ./kiali-operator-1.34.0.tgz came from. Can you confirm you see the same behavior using the published helm chart?

This is the error; I have never seen this:

    Message:               Failed to patch object: b'{"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":" \\"\\" is invalid: patch: Invalid value: \\"{\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"kind\\\\\\":\\\\\\"Namespace\\\\\\",\\\\\\"metadata\\\\\\":{\\\\\\"annotations\\\\\\":{\\\\\\"kubectl.kubernetes.io/last-applied-configuration\\\\\\":\\\\\\"{\\\\\\\\\\\\\\"apiVersion\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"v1\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"kind\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"Namespace\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"metadata\\\\\\\\\\\\\\":{\\\\\\\\\\\\\\"annotations\\\\\\\\\\\\\\":{},\\\\\\\\\\\\\\"name\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"access-psql\\\\\\\\\\\\\\"}}\\\\\\\\n\\\\\\"},\\\\\\"creationTimestamp\\\\\\":\\\\\\"2021-05-11T08:15:22Z\\\\\\",\\\\\\"labels\\\\\\":{\\\\\\"istio-injection\\\\\\":\\\\\\"enabled\\\\\\",\\\\\\"kiali-enabled\\\\\\":true},\\\\\\"managedFields\\\\\\":[{\\\\\\"manager\\\\\\":\\\\\\"kubectl-client-side-apply\\\\\\",\\\\\\"operation\\\\\\":\\\\\\"Update\\\\\\",\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"time\\\\\\":\\\\\\"2021-05-11T08:15:22Z\\\\\\",\\\\\\"fieldsType\\\\\\":\\\\\\"FieldsV1\\\\\\",\\\\\\"fieldsV1\\\\\\":{\\\\\\"f:metadata\\\\\\":{\\\\\\"f:annotations\\\\\\":{\\\\\\".\\\\\\":{},\\\\\\"f:kubectl.kubernetes.io/last-applied-configuration\\\\\\":{}}},\\\\\\"f:status\\\\\\":{\\\\\\"f:phase\\\\\\":{}}}},{\\\\\\"manager\\\\\\":\\\\\\"kubectl-label\\\\\\",\\\\\\"operation\\\\\\":\\\\\\"Update\\\\\\",\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"time\\\\\\":\\\\\\"2021-05-11T11:07:14Z\\\\\\",\\\\\\"fieldsType\\\\\\":\\\\\\"FieldsV1\\\\\\",\\\\\\"fieldsV1\\\\\\":{\\\\\\"f:metadata\\\\\\":{\\\\\\"f:labels\\\\\\":{\\\\\\".\\\\\\":{},\\\\\\"f:istio-injection\\\\\\":{}}}}}],\\\\\\"name\\\\\\":\\\\\\"access-psql\\\\\\",\\\\\\"resourceVersion\\\\\\":\\\\\\"57673\\\\\\",\\\\\\"uid\\\\\\":\\\\\\"7884869d-3bff-4bef-8080-a77b157860e3\\\\\\"},\\\\\\"spec\\\\\\":{\\\\\\"finalizers\\\\\\":[\\\\\\"kubernetes\\\\\\"]},\\\\\\"status\\\\\\":{\\\\\\"phase\\\\\\":\\\\\\"Active\\\\\\"}}\\": v1.Namespace.ObjectMeta: v1.ObjectMeta.Labels: ReadString: expects \\" or n, but found t, error found in #10 byte of ...|enabled\\":true},\\"mana|..., bigger context ...|ls\\":{\\"istio-injection\\":\\"enabled\\",\\"kiali-enabled\\":true},\\"managedFields\\":[{\\"manager\\":\\"kubectl-client-s|...","reason":"Invalid","details":{"causes":[{"reason":"FieldValueInvalid","message":"Invalid value: \\"{\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"kind\\\\\\":\\\\\\"Namespace\\\\\\",\\\\\\"metadata\\\\\\":{\\\\\\"annotations\\\\\\":{\\\\\\"kubectl.kubernetes.io/last-applied-configuration\\\\\\":\\\\\\"{\\\\\\\\\\\\\\"apiVersion\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"v1\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"kind\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"Namespace\\\\\\\\\\\\\\",\\\\\\\\\\\\\\"metadata\\\\\\\\\\\\\\":{\\\\\\\\\\\\\\"annotations\\\\\\\\\\\\\\":{},\\\\\\\\\\\\\\"name\\\\\\\\\\\\\\":\\\\\\\\\\\\\\"access-psql\\\\\\\\\\\\\\"}}\\\\\\\\n\\\\\\"},\\\\\\"creationTimestamp\\\\\\":\\\\\\"2021-05-11T08:15:22Z\\\\\\",\\\\\\"labels\\\\\\":{\\\\\\"istio-injection\\\\\\":\\\\\\"enabled\\\\\\",\\\\\\"kiali-enabled\\\\\\":true},\\\\\\"managedFields\\\\\\":[{\\\\\\"manager\\\\\\":\\\\\\"kubectl-client-side-apply\\\\\\",\\\\\\"operation\\\\\\":\\\\\\"Update\\\\\\",\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"time\\\\\\":\\\\\\"2021-05-11T08:15:22Z\\\\\\",\\\\\\"fieldsType\\\\\\":\\\\\\"FieldsV1\\\\\\",\\\\\\"fieldsV1\\\\\\":{\\\\\\"f:metadata\\\\\\":{\\\\\\"f:annotations\\\\\\":{\\\\\\".\\\\\\":{},\\\\\\"f:kubectl.kubernetes.io/last-applied-configuration\\\\\\":{}}},\\\\\\"f:status\\\\\\":{\\\\\\"f:phase\\\\\\":{}}}},{\\\\\\"manager\\\\\\":\\\\\\"kubectl-label\\\\\\",\\\\\\"operation\\\\\\":\\\\\\"Update\\\\\\",\\\\\\"apiVersion\\\\\\":\\\\\\"v1\\\\\\",\\\\\\"time\\\\\\":\\\\\\"2021-05-11T11:07:14Z\\\\\\",\\\\\\"fieldsType\\\\\\":\\\\\\"FieldsV1\\\\\\",\\\\\\"fieldsV1\\\\\\":{\\\\\\"f:metadata\\\\\\":{\\\\\\"f:labels\\\\\\":{\\\\\\".\\\\\\":{},\\\\\\"f:istio-injection\\\\\\":{}}}}}],\\\\\\"name\\\\\\":\\\\\\"access-psql\\\\\\",\\\\\\"resourceVersion\\\\\\":\\\\\\"57673\\\\\\",\\\\\\"uid\\\\\\":\\\\\\"7884869d-3bff-4bef-8080-a77b157860e3\\\\\\"},\\\\\\"spec\\\\\\":{\\\\\\"finalizers\\\\\\":[\\\\\\"kubernetes\\\\\\"]},\\\\\\"status\\\\\\":{\\\\\\"phase\\\\\\":\\\\\\"Active\\\\\\"}}\\": v1.Namespace.ObjectMeta: v1.ObjectMeta.Labels: ReadString: expects \\" or n, but found t, error found in #10 byte of ...|enabled\\":true},\\"mana|..., bigger context ...|ls\\":{\\"istio-injection\\":\\"enabled\\",\\"kiali-enabled\\":true},\\"managedFields\\":[{\\"manager\\":\\"kubectl-client-s|...","field":"patch"}]},"code":422}\n'

**Reply by bethmage** [reply]
helm chart is pulled from here : https://kiali.org/helm-charts. 
with: helm pull kiali/kiali-operator --version 1.34.0
Anyway the same work with the version v1.32.0 from the above repository

**Reply by jmazzitelli** [reply]
I see a "kiali-enabled: true" label on the namespace that can't be patched - I don't know where that label comes from - it isn't something that Kiali creates. Unsure if this is related.

We need to see the full operator logs. Please zip up the entire operator logs and attach that zip file here - we need to see what the operator was doing.

**Reply by bethmage** [reply]
The label is comes from the kiali_cr :
.spec:
  api:
    namespaces:
      label_selector: kiali-enabled=true
   
Please see: https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L106
as described there:
 If deployment.accessible_namespaces does not have the special value of "**"
 then the Kiali operator will add a new label to all accessible namespaces - that new
 label will be this label_selector.   
 which is my case deployment.accessible_namespaces  value is a regx not '**'
 The kiali-operator log:
[kiali-operator.tar.gz](https://github.com/kiali/kiali/files/6461364/kiali-operator.tar.gz)

**Comment by jmazzitelli** [comment]
I think it has to do with that custom "kiali-enabled: true" label.

The error message `ReadString: expects \\" or n, but found t, error found in #10 byte of ...|enabled\\":true}` indicates there is a problem with that label. I believe it is because you added this label with a boolean value `true` when you should be using it as a string (wrapped in quotes) -- `"true"`. Kubernetes labels must be a string, not a boolean, see https://github.com/kubernetes/kubernetes/issues/23251

Is there a reason you are customizing this label selector? You shouldn't need to set that unless you have a reason to. Kiali will define it for you when it needs (see: https://github.com/kiali/kiali-operator/blob/master/roles/default/kiali-deploy/tasks/main.yml#L503).

But if you really want to set it yourself, it needs to be something other than the boolean value `true`.

**Reply by jmazzitelli** [reply]
Looking at your operator logs, what I mention above is definitely the problem. Error occurs in the task:

```
TASK [Create additional Kiali label on all accessible namespaces] ******************************** 
```

I recommend taking out that label_selector in your CR - you don't need to set it unless you have a very specific reason for wanting that label. If you still want it, the "kiali-enabled" label must be something OTHER THAN the boolean `true` (as you can see, Kubernetes does not allow you to set values of labels to boolean values - they must be strings).

**Reply by bethmage** [reply]
The reason to customizing this label was, just to which namespaces are really selected by kiali. 
 Just now i tested the installation without the label_selector: kiali-enabled=true (removed)  
 So it is **working now**  Thank you so much.
 
 What i did not understand is, WHY it is workling with the version : V1.32.0

**Reply by jmazzitelli** [reply]
you will get a label set - as those comments you pointed to earlier mention:

https://github.com/kiali/kiali-operator/blob/master/deploy/kiali/kiali_cr.yaml#L102-L105

```
# Note that if you do not set this but deployment.accessible_namespaces does not have the
# special "all namespaces" value of "**" then this label_selector will be set
# to a default value of "kiali.io/member-of=<deployment.namespace>" where
# <deployment.namespace> is the namespace where Kiali is to be installed.
``` 

So you will see a label on the namespaces kiali is accessing called `kiali.io/member-of` and the value will be the control plane namespace (typically `istio-system`).

**Reply by jmazzitelli** [reply]
> What i did not understand is, WHY it is workling with the version : V1.32.0

Nothing changed in this area since v1.32 that I recall. Must have been some other error you were hitting unrelated to this "boolean label value" problem.

**Reply by bethmage** [reply]
OK. Thanks again
